File: backend/SRMS.API/Models/EmergencySession.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace SRMS.API.Models;

[Table("emergency_sessions")]
public class EmergencySession
{
    [Key]
    [Column("session_id")]
    public Guid SessionId { get; set; }

    [Required]
    [Column("driver_id")]
    public Guid DriverId { get; set; }

    [Required]
    [MaxLength(100)]
    [Column("vehicle_type")]
    public string VehicleType { get; set; } = string.Empty;

    [Required]
    [MaxLength(50)]
    [Column("status")]
    public string Status { get; set; } = string.Empty;

    [Column("selected_route_id")]
    public Guid? SelectedRouteId { get; set; }

    [Column("created_at")]
    public DateTime CreatedAt { get; set; } = DateTime.UtcNow;

    [Column("updated_at")]
    public DateTime UpdatedAt { get; set; } = DateTime.UtcNow;

    // Navigation properties
    [ForeignKey(nameof(SelectedRouteId))]
    public virtual TrafficRoute? SelectedRoute { get; set; }

    public virtual ICollection<SignalPreemptionLog> SignalPreemptionLogs { get; set; } = new List<SignalPreemptionLog>();
}

File: mobile/lib/models/emergency_session.dart
class EmergencySession {
  final String sessionId;
  final String driverId;
  final String vehicleType;
  final String status;
  final String? selectedRouteId;
  final DateTime createdAt;
  final DateTime updatedAt;

  EmergencySession({
    required this.sessionId,
    required this.driverId,
    required this.vehicleType,
    required this.status,
    this.selectedRouteId,
    required this.createdAt,
    required this.updatedAt,
  });

  factory EmergencySession.fromJson(Map<String, dynamic> json) {
    return EmergencySession(
      sessionId: json['sessionId'] as String,
      driverId: json['driverId'] as String,
      vehicleType: json['vehicleType'] as String,
      status: json['status'] as String,
      selectedRouteId: json['selectedRouteId'] as String?,
      createdAt: DateTime.parse(json['createdAt'] as String),
      updatedAt: DateTime.parse(json['updatedAt'] as String),
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'sessionId': sessionId,
      'driverId': driverId,
      'vehicleType': vehicleType,
      'status': status,
      'selectedRouteId': selectedRouteId,
      'createdAt': createdAt.toIso8601String(),
      'updatedAt': updatedAt.toIso8601String(),
    };
  }
}
